src/test/java/org/jenkinsci/plugins/badge/extensions/SpecialValueParameterResolverExtensionTest.java

```diff
@@ -73,4 +75,37 @@ void testMultipleParameters() {
                 extension.resolve(mockRun, "buildId buildNumber duration startTime displayName description");
         assertThat(actualParameter, is("1234 1234 23:35 23:35 display name run description"));
     }
+
+    @Test
+    void resolveEmptyStringParameter() {
+        String actualParameter = extension.resolve(mockRun, "");
+        assertThat(actualParameter, is(""));
+    }
+
+    @Test
+    void resolveWithNullParameter() {
+        String actualParameter = extension.resolve(mockRun, null);
+        assertThat(actualParameter, is((String) null));
+    }
+
+    @Test
+    void unknownParameterShouldNotBeResolved() {
+        String actualParameter = extension.resolve(mockRun, "unknown");
+        assertThat(actualParameter, is("unknown"));
+    }
+
+    @Test
+    void testResolveWithNoMatchingParameters() {
+        String actualParameter = extension.resolve(mockRun, "noMatchingParameters");
+        assertThat(actualParameter, is("noMatchingParameters"));
+    }
+
+    @Test
+    void testResolveWhenJobHasNoLastBuild() {
+        Job<?, ?> job = mock(Job.class);
+        when(job.getLastBuild()).thenReturn(null);
+
+        String parameter = "${buildId}${buildNumber}";
+        assertEquals("${buildId}${buildNumber}", extension.resolve(job, parameter));
```

Comment:
The `assertEquals` method provides a less helpful message on failure than `assertThat` from Hamcrest.  Let's follow the pattern of using Hamcrest assertions in this file.  Since the `parameter` is declared with that value, let's assert that is the correct value rather than copying the string.
```suggestion
        assertThat(extension.resolve(job, parameter), is(parameter));
```